payments/product-inventory/src/main/java/com/appcoins/product_inventory/ProductInventoryRepository.kt

```diff
@@ -53,55 +69,97 @@ internal class ProductInventoryRepositoryImpl(
   override suspend fun getPurchases(
     packageName: String,
     ewt: String,
-  ): List<PurchaseInfoData> = restClient
-    .get(
-      path = "productv2/8.20230522/applications/$packageName/inapp/consumable/purchases",
-      header = mapOf("authorization" to "Bearer $ewt"),
-      query = mapOf(
-        "type" to "INAPP",
-        "state" to "PENDING",
-        "sku" to null,
-      ),
-    )
-    ?.jsonToPurchasesResponse()!!
-    .items
-    .map { it.toPurchaseInfoData(packageName) }
+  ): List<PurchaseInfoData> = runCatching {
+    restClient
+      .get(
+        path = "productv2/8.20230522/applications/$packageName/inapp/consumable/purchases",
+        header = mapOf("authorization" to "Bearer $ewt"),
+        query = mapOf(
+          "type" to "INAPP",
+          "state" to "PENDING",
+          "sku" to null,
+        ),
+      )
+      ?.jsonToPurchasesResponse()!!
+      .items
+      .map { it.toPurchaseInfoData(packageName) }
+  }.getOrElse { throw handleException(it) }
 
   override suspend fun consumePurchase(
     domain: String,
     uid: String,
     authorization: String,
     payload: String?,
-  ): Boolean = restClient.post(
-    path = "productv2/8.20230522/applications/$domain/inapp/purchases/$uid/consume",
-    header = mapOf("authorization" to "Bearer $authorization"),
-    query = mapOf("payload" to payload)
-  ).let { true }
+  ): Boolean = runCatching {
+    restClient.post(
+      path = "productv2/8.20230522/applications/$domain/inapp/purchases/$uid/consume",
+      header = mapOf("authorization" to "Bearer $authorization"),
+      query = mapOf("payload" to payload)
+    ).let { true }
+  }.getOrElse { throw handleException(it) }
+
+  private fun handleException(exception: Throwable): Throwable {
```

Comment:
It does not handle but transforms or maps 😄 
Let's rename it to `mapException` mb?